src/cmakeTools.ts

```diff
@@ -2220,14 +2220,17 @@ export class CMakeTools implements api.CMakeToolsAPI {
         }
 
         const userConfig = this.workspaceContext.config.debugConfig;
-        const terminal = await this.createTerminal(executable);
+        const terminal: vscode.Terminal = await this.createTerminal(executable);
 
         let executablePath = shlex.quote(executable.path);
 
         if (process.platform === 'win32') {
             executablePath = executablePath.replace(/\\/g, "/");
-
-            if (vscode.env.shell.indexOf("PowerShell") > 0) {
+            const launchBehavior = this.workspaceContext.config.launchBehavior.toLowerCase();
+            const launchTerminalPath = (terminal.creationOptions as vscode.TerminalOptions).env![this.launchTerminalPath];
+            if (vscode.env.shell.indexOf("PowerShell") > 0
```

Comment:
The indexof checks should always be against `launchTerminalPath`, not `vscode.env.shell`.
We should do case-insensitive checks, so please lowercase the path first.
We shouldn't need to do anything different based on the `launchBehavior`.